### PR TITLE
Python 3.X Support

### PR DESCRIPTION
**Pull request by ax3l**
Follow up to #33: Adds a Python 3.X test to CI via travis. ~~Currently, due to errors in it we allow a 3.X failure without marking the PR as invalid. (Nevertheless, we might want to add python 3 support.)~~

At least for what is covered by the tests, this PR adds Python 3.X support to openPMD-viewer (alongside its Python 2 support).

Additionally, some missing end-of-file empty newlines are added for better diff readability.

### CODE REVIEW

File: tests/test_tutorials.py [hunk 1]
```diff
@@ -31,7 +34,13 @@ def test_tutorials():
         script_name = notebook_name[:-6] + '.py'
         os.system( 'ipython nbconvert --to=python %s' %notebook_name )
         clean_ipython_features( script_name )
-        execfile( script_name )
+        try:
+            exec( open(script_name).read() )
```
Comment (ax3l) [inline tests/test_tutorials.py]: Note: I am still *sourcing* the file which is probably not ideal and can have side-effects.

Using `os.system()` and checking its return status is probably the better choice for a follow-up refactoring.

File: tests/test_tutorials.py [hunk 2]
```diff
@@ -54,12 +63,12 @@ def clean_ipython_features( script_name ):
 
         # Replace the lines that activate matplotlib in a notebook
         # by a line that selects the PS backend
-        if re.match("[ ]*get_ipython\(\)\.magic\(u'matplotlib",
+        if re.match("[ ]*get_ipython\(\)\.magic\((.*)'matplotlib",
```
Comment (ax3l) [inline tests/test_tutorials.py]: utf8 just became standard in python3 so we expect *zero or more* chars here

**Comment by RemiLehe** [comment]
Thanks for the code improvements ! And thanks a lot for debugging the Python 3 version !!

I have a few minor changes, which I'll submit via a PR to this branch.

**Comment by ax3l** [comment]
sweet, thanks I will wait for them! :)

**Comment by RemiLehe** [comment]
Here it is : 
https://github.com/ax3l/openPMD-viewer/pull/1

**Comment by ax3l** [comment]
merged and updated (I re-added the "print script on error" functionality)

**Comment by RemiLehe** [comment]
Great ! Thanks for the modifications.

**Pull request merged**